### ISSUE TITLE
Error basic::Singleton 5

### ISSUE THREAD

**Issue by AnantaYudica** [AnantaYudica/basic#48]
```
#include "Singleton.h"
#define USING_BASIC_TEST_MEMORY
#include "Test.h"

BasicTest& bt = BasicTest::ms_instance;

class Value
{
private:
    int m_value;
public:
    Value() : m_value(0) {}
    Value(const int& val) : m_value(val){}
public:
    int& GetValue() {return m_value;}
    const int& GetValue() const {return m_value;} 
};

class C :
    public Value,
    public basic::Singleton<C*>
{
    friend basic::Singleton<C*>;
private:
    static void Construct(void * p)
    {
        new (p) C*(new C);
    }
    static void Destroy(C** p)
    {
        delete *p;
    }
    C() {}
};


int main()
{
    C::ConstructInstance();
}

```
`macro/MemberDefinition.h:57:76: error: 'C*' is not a class, struct, or union type`

**Comment by AnantaYudica** [comment]
Fixed in v0.1-beta-2.

**Comment by AnantaYudica** [comment]
Reopen, not fixed in v0.1-beta-2, Related with bug #46
